wayland-cursor/src/lib.rs

```diff
@@ -177,6 +194,63 @@ impl CursorTheme {
         }
     }
 
+    /// try to load data from theme, else, it will try to load from fallback data
+    ///
+    /// This method returns [`None`] if this cursor is not provided either by the theme, or by one of its parents,
+    /// and the data by fallback function cannot parse it
+    /// fallback function will receive a enum which discribe the error, you can print some messages
+    /// or do some other things before return the finally fallback data
+    pub fn get_cursor_with_default_data<F>(&mut self, name: &str, fallback: F) -> Option<&Cursor>
+    where
```

Comment:
I was rather thinking of defining the fallback closure one and for all as a standalone method of `CursorTheme`, so that the rest of the API does not need to change at all.

Or is having an independent closure for each time you retrieve the cursor, with all that error-processing machinery something that you need for your uses?

Reply:
Emm, what do you mean , make the function out of CursorTheme?

Reply:
My idea was to add to `CursorTheme` a method like this:

```rust
pub fn set_fallback(fb: impl FnMut(&str) -> Cow<'static, [u8]>) { /* ... */ }
```

that stores a fallback closure inside the `CursorTheme`, that would then be used internally by `get_cursor()` if the system theme does not provide the requested cursor.

Reply:
then I think pass a fb is needless, and maybe here need to return a result

Reply:
ha, I see